src/resolve/resolvectl.c

```diff
@@ -162,6 +162,15 @@ static void print_source(uint64_t flags, usec_t rtt) {
                yes_no(flags & SD_RESOLVED_AUTHENTICATED),
                yes_no(flags & SD_RESOLVED_CONFIDENTIAL),
                ansi_normal());
+
+        if ((flags & (SD_RESOLVED_FROM_MASK|SD_RESOLVED_SYNTHETIC)) != 0)
+                printf("%s-- Data from:%s%s%s%s%s\n",
+                       ansi_grey(),
+                       FLAGS_SET(flags, SD_RESOLVED_SYNTHETIC) ? " synthetic" : "",
+                       FLAGS_SET(flags, SD_RESOLVED_FROM_CACHE) ? " cache" : "",
+                       FLAGS_SET(flags, SD_RESOLVED_FROM_ZONE) ? " zone" : "",
+                       FLAGS_SET(flags, SD_RESOLVED_FROM_TRUST_ANCHOR) ? " trust-anchor" : "",
+                       FLAGS_SET(flags, SD_RESOLVED_FROM_NETWORK) ? " network" : "");
```

Comment:
So the reason we show the protocol always is that it does show where stuff was routed, even if in the end the answer was synthesized locally. i.e. stuff such as the trust anchor is DNS stuff, not mDNS/LLMNR stuff, and that's relevant. And the zones are mdnS/LLMNR stuff, there are two distinct zones for each, since those two register RRs on the local network after all, hence it is relevant to show the protocol for them too.

Reply:
Then at least this shouldn't say "protocol DNS". For synthesized names, there is no "DNS protocol" used anywhere. Also saying "acquire via" misleads even more.

I guess we want:
```
-- Query was routed to {DNS|mDNS|LLMNR} (maybe some more info here)
-- Information was {received via ...|synthesized from ...}
-- Data is {authenticated|non-authenticated}
```